tests/components/apache_kafka/test_apache_kafka.py

```diff
@@ -4,53 +4,51 @@
 import pytest
 
 import homeassistant.components.apache_kafka as apache_kafka
+from homeassistant.const import STATE_ON
 from homeassistant.core import split_entity_id
-from homeassistant.helpers.entityfilter import FILTER_SCHEMA
 from homeassistant.setup import async_setup_component
 
-from tests.async_mock import MagicMock, Mock, patch
+from tests.async_mock import MagicMock, patch
 
 APACHE_KAFKA_PATH = "homeassistant.components.apache_kafka"
+PRODUCER_PATH = f"{APACHE_KAFKA_PATH}.AIOKafkaProducer"
+MIN_CONFIG = {
+    "ip_address": "localhost",
+    "port": 8080,
+    "topic": "topic",
+}
+FilterTest = namedtuple("FilterTest", "id should_pass")
+MockKafkaClient = namedtuple("MockKafkaClient", "init start send_and_wait")
 
 
-@pytest.fixture(autouse=True, name="mock_client_methods")
-def mock_client_methods_fixture():
+@pytest.fixture(name="mock_client")
+def mock_client_fixture():
     """Mock the apache kafka client."""
-    producer_path = f"{APACHE_KAFKA_PATH}.AIOKafkaProducer"
-    with patch(f"{producer_path}.start") as start:
-        with patch(f"{producer_path}.stop") as stop:
-            with patch(f"{producer_path}.send_and_wait") as send_and_wait:
-                yield (start, stop, send_and_wait)
-
-
-@pytest.fixture(autouse=True)
-def mock_bus_and_json(hass, monkeypatch):
-    """Mock the event bus listener and os component."""
-    hass.bus.async_listen = MagicMock()
-    monkeypatch.setattr(
-        f"{APACHE_KAFKA_PATH}.json.dumps", Mock(return_value=MagicMock())
-    )
+    with patch(f"{PRODUCER_PATH}.start") as start:
+        with patch(f"{PRODUCER_PATH}.send_and_wait") as send_and_wait:
+            with patch(f"{PRODUCER_PATH}.__init__", return_value=None) as init:
```

Comment:
```suggestion
    with patch(f"{PRODUCER_PATH}.start") as start,  patch(f"{PRODUCER_PATH}.send_and_wait") as send_and_wait, patch(f"{PRODUCER_PATH}.__init__", return_value=None) as init:
```

Reply:
Oh nice, didn't know you could do that, will do.  

FYI I tracked down the pipeline error with azure and have a fix for that so will commit momentarily